projects/lux/src/lib/geolocation/geolocation.component.ts

```diff
@@ -0,0 +1,167 @@
+import { Component, Input, Output, EventEmitter, OnInit } from '@angular/core';
+import { FormControl, Validators, ValidatorFn } from '@angular/forms';
+import { ModalService } from '../modal/modal.service';
+import { Geopoint } from './geopoint';
+
+@Component({
+  selector: 'lux-geolocation',
+  templateUrl: './geolocation.component.html',
+  styleUrls: ['./geolocation.component.scss']
+})
+export class GeolocationComponent implements OnInit {
+  static idCounter = 0;
+
+  @Input()
+  public minLatitude = -90;
+  @Input()
+  public maxLatitude = 90;
+  @Input()
+  public minLongitude = -180;
+  @Input()
+  public maxLongitude = 180;
+  @Input()
+  public step = 0.00001; // 0.00001 degrees = 1.11 meters
+  private _disabled: string | boolean;
+  private _required: boolean;
+  private _value: any;
+  public latitudeValue: number = null;
+  public longitudeValue: number = null;
+  public latitudeValidators: ValidatorFn[] = [];
+  public longitudeValidators: ValidatorFn[] = [];
+  public latitudeFormControl = new FormControl(this.latitudeValue);
+  public longitudeFormControl = new FormControl(this.longitudeValue);
+
+  get className(): string {
+    return this.checkClassName();
+  }
+
+  @Input() public inputId: string;
+  @Input('aria-label') public ariaLabel: string;
+  @Input() public readonly: boolean | null = null;
+
+  @Input()
+  set disabled(v: string | boolean) {
+    v = typeof v === 'string' ? true : v;
+    this._disabled = v;
+  }
+  get disabled(): string | boolean {
+    return this._disabled;
+  }
+
+  @Input()
+  set required(v: boolean) {
+    this._required = v;
+    this.updateValidators([Validators.required], [Validators.required]);
+  }
+  get required(): boolean {
+    return this._required;
+  }
+
+  @Input()
+  set value(v: Geopoint) {
+    if (v === this._value) {
+      return; // prevent events when there is no changes
+    }
+    if (v.coordinates && v.coordinates.length === 2) {
+      this._value = v;
+      this.latitudeValue = +v.coordinates[1];
+      this.longitudeValue = +v.coordinates[0];
+      this.latitudeFormControl.setValue(this.latitudeValue);
+      this.longitudeFormControl.setValue(this.longitudeValue);
+    }
+    this.valueChange.emit(v);
+  }
+  get value(): Geopoint {
+    this._value = {
```

Comment:
si this.longitudeFormControl.value o this.latitudeFormControl.value son undefined -> deberiamos devolver un `return undefined;`